src/Messages.sol

```diff
@@ -159,8 +161,8 @@ library ConnectorMessages {
      * 47-78: amount (uint256 = 32 bytes)
      * 
      */
-    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint256 amount) internal pure returns (bytes memory) {
-        return abi.encodePacked(uint8(Call.Transfer), poolId, trancheId, user, amount);
+    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint256 amount, uint32 destinationDomain) internal pure returns (bytes memory) {
```

Comment:
This is not correct since `domain` is not `uint32`. Either we receive `bytes9` here and expect the caller of `formatTransfer` to first call `formatDomain` first, or we overload `formatTransfer for 1) `(Domain domain) and 2) Domain domain, uint64 domainId)` and call the right `formatDomain` internally, respectively.

Reply:
I think changing it to bytes9 is the cleanest solution